keystorage/aws/src/main/java/tech/pegasys/signers/aws/AwsSecretsManager.java

```diff
@@ -68,6 +81,54 @@ public Optional<String> fetchSecret(final String secretName) {
     }
   }
 
+  private ListSecretsIterable listSecrets(
+      final List<String> tagKeys, final List<String> tagValues) {
+    final ListSecretsRequest.Builder listSecretsRequestBuilder = ListSecretsRequest.builder();
+    if (!tagKeys.isEmpty()) {
+      listSecretsRequestBuilder.filters(
+          Filter.builder().key(FilterNameStringType.TAG_KEY).values(tagKeys).build());
+    }
+    if (!tagValues.isEmpty()) {
+      listSecretsRequestBuilder.filters(
+          Filter.builder().key(FilterNameStringType.TAG_VALUE).values(tagValues).build());
+    }
+    return secretsManagerClient.listSecretsPaginator(listSecretsRequestBuilder.build());
+  }
+
+  public <R> Collection<R> mapSecrets(
+      final List<String> tagKeys,
+      final List<String> tagValues,
+      final BiFunction<String, String, R> mapper) {
+    final Set<R> result = ConcurrentHashMap.newKeySet();
+    listSecrets(tagKeys, tagValues)
+        .iterator()
+        .forEachRemaining(
+            listSecretsResponse -> {
+              listSecretsResponse
+                  .secretList()
+                  .parallelStream()
+                  .forEach(
+                      secretEntry -> {
+                        try {
+                          final String secretValue = fetchSecret(secretEntry.name()).get();
```

Comment:
This doesn't handle the case where a secret isn't found. The mapper should only try and map the value if it exists. Not that likely but it's possible that the secret no longer exists when we go to fetch the value. In the case we get an empty result I think we can just log a warning and skip it like we do with other errors.